sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/transaction/OutboundTransactionManagerImpl.kt

```diff
@@ -0,0 +1,83 @@
+package cash.z.ecc.android.sdk.internal.transaction
+
+import cash.z.ecc.android.sdk.internal.Twig
+import cash.z.ecc.android.sdk.internal.model.EncodedTransaction
+import cash.z.ecc.android.sdk.model.Account
+import cash.z.ecc.android.sdk.model.TransactionRecipient
+import cash.z.ecc.android.sdk.model.UnifiedSpendingKey
+import cash.z.ecc.android.sdk.model.Zatoshi
+import co.electriccoin.lightwallet.client.LightWalletClient
+import co.electriccoin.lightwallet.client.model.Response
+
+@Suppress("TooManyFunctions")
+internal class OutboundTransactionManagerImpl(
+    internal val encoder: TransactionEncoder,
+    private val service: LightWalletClient
+) : OutboundTransactionManager {
+
+    override suspend fun encode(
+        usk: UnifiedSpendingKey,
+        amount: Zatoshi,
+        recipient: TransactionRecipient,
+        memo: String,
+        account: Account
+    ): EncodedTransaction {
+        return when (recipient) {
+            is TransactionRecipient.Account -> {
+                encoder.createShieldingTransaction(
+                    usk,
+                    recipient,
+                    memo.toByteArray()
+                )
+            }
+            is TransactionRecipient.Address -> {
+                encoder.createTransaction(
+                    usk,
+                    amount,
+                    recipient,
+                    memo.toByteArray()
+                )
+            }
+        }
+    }
+
+    override suspend fun submit(encodedTransaction: EncodedTransaction) {
+        when (val response = service.submitTransaction(encodedTransaction.raw.byteArray)) {
+            is Response.Success -> {
+                Twig.debug { "SUCCESS: submit transaction completed with response: ${response.result}" }
+            }
+
+            is Response.Failure -> {
```

Comment:
Should we also report the submission failure result to the caller, or is it done another way?